components/Nav/Navbar/index.tsx

```diff
@@ -1,36 +1,204 @@
+import HamburgerMenu from '~/components/Nav/Navbar/MobileMenus/HamburgerMenu';
+import LauncherMenu from '~/components/Nav/Navbar/MobileMenus/LauncherMenu';
+import { PopoutButtons } from '~/components/Nav/Navbar/Popouts/Buttons';
+import TracerNavLogo from '~/components/Nav/Navbar/TracerNavLogo';
+import { NavContext, NavContextProvider } from '~/context/NavContext';
 import { useStore } from '~/store/main';
 import { selectWeb3Info } from '~/store/Web3Slice';
 
-import HeaderSiteSwitcher from './HeaderSiteSwitcher';
+import { classNames } from '~/utils/helpers';
 import AccountDropdown from './AccountDropdown';
-import MobileMenu from './MobileMenu';
+import MobileMenu from './MobileMenus/MobileNav';
 import NetworkDropdown from './NetworkDropdown';
-import { classNames } from '~/utils/helpers';
-import ThemeSwitcher from './ThemeSwitcher';
-import VersionToggle from './VersionToggle';
-
-import RevisitOnboard from '/public/img/general/onboard-revisit.svg';
+import HelpIconSVG from '/public/img/general/onboard-revisit.svg';
 
 const NavBar: React.FC<{
     setShowOnboardModal?: React.Dispatch<React.SetStateAction<boolean>>;
 }> = ({ setShowOnboardModal }) => {
+    return (
+        <NavContextProvider>
+            <NavBarContent setShowOnboardModal={setShowOnboardModal} />
+        </NavContextProvider>
+    );
+};
+
+const NavBarContent: React.FC<{
+    setShowOnboardModal?: React.Dispatch<React.SetStateAction<boolean>>;
+}> = ({ setShowOnboardModal }) => {
+    const { navMenuOpen, setNavMenuOpen, launcherMenuOpen, setLauncherMenuOpen } = useContext(NavContext);
+    const [navBackdrop, setNavBackdrop] = useState<boolean>(true);
+    const routes = useRouter().asPath.split('/');
+    const route = routes[1];
+    const { account, network } = useStore(selectWeb3Info, shallow);
+
+    const listItemStyles = 'flex';
+    const linkStyles =
+        'flex transition-all items-center px-4 py-2 text-base cursor-pointer whitespace-nowrap dark:text-white text-tracer-650';
+    const selectedStyles = 'font-bold';
+
+    const handleScroll = (scrollContainer: HTMLDivElement) => {
+        if ((!!scrollContainer && scrollContainer.scrollTop >= 1) || window.innerWidth < 1024) {
+            setNavBackdrop(true);
+        } else {
+            setNavBackdrop(false);
+        }
+    };
+
+    useEffect(() => {
+        const scrollContainer = document.getElementById('__next') as HTMLDivElement;
+        handleScroll(scrollContainer);
+        scrollContainer && scrollContainer.addEventListener('scroll', () => handleScroll(scrollContainer));
+        return () => {
+            scrollContainer && scrollContainer.removeEventListener('scroll', () => handleScroll(scrollContainer));
+        };
+    }, [handleScroll]);
+
+    const handleMenuOpen = () => {
+        if (launcherMenuOpen) {
+            setLauncherMenuOpen(false);
+            setTimeout(() => {
+                setNavMenuOpen(true);
+            }, 550);
```

Comment:
same question here about using computed/manually defined animation durations from a single source if possible

Reply:
used constant variable in place of this